src/lib/mongoBackend/MongoGlobal.cpp

```diff
@@ -2718,26 +1779,18 @@ bool processOnChangeConditionForSubscription
   const std::string&               tenant,
   const std::string&               xauthToken,
   const std::vector<std::string>&  servicePathV,
-  const std::string&               qFilter
+  Restriction*                     resP
 )
 {
   std::string                   err;
   NotifyContextRequest          ncr;
-  Restriction                   res;
   ContextElementResponseVector  rawCerV;
 
-  /* Include scopes for entitiesQuery() if q filter is not empty */
-  if (qFilter != "")
-  {
-    Scope* sc = new Scope(SCOPE_TYPE_SIMPLE_QUERY, qFilter);
-    res.scopeVector.push_back(sc);
-  }
-
-  if (!entitiesQuery(enV, attrL, res, &rawCerV, &err, true, tenant, servicePathV))
+  if (!entitiesQuery(enV, attrL, *resP, &rawCerV, &err, true, tenant, servicePathV))
   {
     ncr.contextElementResponseVector.release();
     rawCerV.release();
-    res.release();
+    resP->release();
```

Comment:
`resP` is provides by the caller of processOnChangeConditionForSubscription() functions. For processOnChangeConditionForSubscription() this is like a read-only parameter.

Thus, processOnChangeConditionForSubscription() should be a good fellow :) and not releasing this (as processOnChangeConditionForSubscription() doesn't know if it caller will need resP this for something else).